### DISCUSSION TITLE
Can Skywalking intercept one method of a class mutiple times?

### DISCUSSION THREAD

**Discussion by lingli0219** [apache/skywalking#9552] [Q&A] [answered]
I want to intercept one mthod of a class twice for different purposes. But I found it can only be intercepted once. Can Skywalking intercept one method of a class mutiple times?

**Comment by pg-yang** [comment, marked as answer]
Skywalking couldn't do this , And shouldn't to do . Skywalking is used for APM  . If you want to intercept method for your goal , you  could  add your  own  agent  .
